addons/account/models/account_move.py

```diff
@@ -1344,7 +1344,9 @@ def _check_fiscalyear_lock_date(self):
 
     @api.multi
     def _check_tax_lock_date(self):
-        if not self:
+        moves = self.filtered(lambda move: any(line.tax_repartition_line_id or line.tax_ids for line in move.line_ids))
```

Comment:
What about tax adjustments?
=> We should also check the tags.

Change to check the tax lock date if the aml hax tax_ids or tax_repartition_line_id, or if some of its tag_ids have applicability=='taxes'

Reply:
to be fixed on saas-12.3